addons/delivery/models/sale_order.py

```diff
@@ -48,7 +48,7 @@ def set_delivery_line(self, carrier, amount):
         self._remove_delivery_line()
 
         for order in self:
-            order._create_delivery_line(carrier, amount, price_unit_in_description=self.carrier_id.invoice_policy == 'real')
+            order._create_delivery_line(carrier, amount)
```

Comment:
write commit message explaining that the carrier can be different than the one set on the order, and so we should use the correct setting of the carrier on which we are calling the request.